pna.p4

```diff
@@ -566,15 +577,26 @@ extern bool is_host_port (in PortId_t p);
 // false.
 extern bool is_net_port (in PortId_t p);
 
+// struct `pna_main_input_metadata_t` contains fields initialized for
+// all packets that are input to main control.
 struct pna_main_input_metadata_t {
-    // common fields initialized for all packets that are input to main
-    // parser, regardless of direction.
+    // `recirculated` has the same value as the field with the same
+    // name in struct `pna_main_parser_input_metadata_t` did when this
+    // packet was parsed.
     bool                     recirculated;
+
+    // `recirc_data` has the same value as the field with the same
+    // name in struct `pna_main_parser_input_metadata_t` did when this
+    // packet was parsed.
+    bit<32>                  recirc_data;
```

Comment:
Propose changing bit<32> to type parameter T, like so:
```
struct pna_main_input_metdata_t<T> {
      // other fields here
      T recirc_data;
}
```
also hopefully the package definitions below can be used to force T to be the same between parser and main control.